File: PrimeNumbersPython/PrimeNumber.py
#  python 3.8.3

import numba
from datetime import datetime


@numba.njit()
def prime(n):
    n += 1
    PrimeNumbers = []  # Numba refers arry element type from append function below
    for i in range(2, n):
        c = 2
        while True:

            if i % c == 0:
                break

            c += 1

            if c == i:
                PrimeNumbers.append(i)
                break

    # print(PrimeNumbers)
    return PrimeNumbers


TestCount = 10

for i in range(TestCount):

    start = datetime.now()
    prime(100000)

    print(f"ExCount #{i+1} | Execution time = {(datetime.now()-start).total_seconds()}s")

File: PrimeNumbersGo/PrimeNumber.go
// golang 1.14.4

package main

import (
	"fmt"
	"time"
)

func prime(n int) []int {
	n++
	PrimeNumbers := []int{}
	for i := 2; i < n; i++ {

		c := 2
		for {

			if i%c == 0 {

				break
			}

			c++

			if c == i {

				PrimeNumbers = append(PrimeNumbers, i)
				break

			}

		}

	}
	// fmt.Println(PrimeNumbers)
	return PrimeNumbers
}

func main() {

	TestCount := 10

	for i := 0; i < TestCount; i++ {
		start := time.Now()
		prime(100000)

		fmt.Printf("ExCount #%v | Execution time = %vs\n", i+1, time.Now().Sub(start).Seconds())

	}

}
